pkg/cache/tas_flavor_snapshot.go

```diff
@@ -540,16 +550,37 @@ func (s *TASFlavorSnapshot) findTopologyAssignment(
 	podSetTolerations := info.Tolerations
 	podSetNodeSelectors := info.NodeSelector
 	count := tasPodSetRequests.Count
+
+	// If chunk topology is not requested then we can assume that chunk is a single pod
+	chunkSize, reason := getChunkSizeWithDefault(tasPodSetRequests.PodSet.TopologyRequest, 1)
+	if len(reason) > 0 {
+		return nil, reason
+	}
+
 	required := isRequired(tasPodSetRequests.PodSet.TopologyRequest)
-	key := s.levelKeyWithImpliedFallback(&tasPodSetRequests)
+
+	topologyKey := s.levelKeyWithImpliedFallback(&tasPodSetRequests)
```

Comment:
Consider pulling out this rename to the same preparatory PR (where you define sorting functions) to reduce this diff size

Reply:
Done! I've extracted the rename to other PR :)